src/ResourceManager/PowerBIDedicated/Commands.PowerBIDedicated/Commands/GetAzureRmPowerBIEmbeddedCapacity.cs

```diff
@@ -0,0 +1,54 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.PowerBIDedicated.Models;
+using Microsoft.Azure.Management.PowerBIDedicated;
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Management.PowerBIDedicated.Models;
+
+namespace Microsoft.Azure.Commands.PowerBIDedicated
+{
+    [Cmdlet(VerbsCommon.Get, "AzureRmPowerBIEmbeddedCapacity"),
+     OutputType(typeof(List<AzurePowerBIDedicated>))]
+    public class GetAzurePowerBIEmbeddedCapacity : PowerBIDedicatedCmdletBase
+    {
+        [Parameter(Position = 0,
+            ValueFromPipelineByPropertyName = true, Mandatory = false,
+            HelpMessage = "Name of resource group under which the user want to retrieve the capacity.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Position = 1, ValueFromPipelineByPropertyName = true,
+            Mandatory = false, HelpMessage = "Name of a specific capacity.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        public override void ExecuteCmdlet()
```

Comment:
@selasagi please add another parameter set that allows users to pass just a `ResourceId` to this cmdlet from the generic `Get/Find-AzureRmResource` cmdlet via piping. This should be in a separate parameter from `Name` and `ResourceGroupName` and should have the `ValueFromPipelineByPropertyName = true` property.

Reply:
Done